blockchain/validation.go

```diff
@@ -69,8 +121,51 @@ func (bc *BlockChain) ValidTransaction(t *Transaction) (bool, TransactionCode) {
 	return true, ValidTransaction
 }
 
+// ValidCloudBase returns true if a transaction is a valid CloudBase transaction
+// and false otherwise
+func ValidCloudBase(t *Transaction) (bool, CloudBaseTransactionCode) {
```

Comment:
Isn't the point of ValidCloudBase to validate the cloud base transaction before it is added to a block? The block only stores the information of the address of the recipient of the block reward in the cloudbase transaction itself. Therefore shouldnt it just take in the address and the block reward in instead of the block?